### PR TITLE
fix: Improve IAM Report Error Handling

### LINKED ISSUE #1001: Iam users GenerateCredentialReport has paralel reports capacity that is not accounted
### Describe the bug

Sometimes users face next error during credential report generation:
```
api error LimitExceeded: No more capacity to launch new report generation tasks
```

### Expected Behavior

this error should not happen and should not fail the fetch

### Steps to Reproduce

run `cloudquery fetch` for iam users 

### Possible Solution

add a logic that waits for some time when this error happens and try to generate a report again.

### Provider and CloudQuery version

aws 0.12.4

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by bbernays**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [ ] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [ ] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

File: resources/services/iam/users.go [hunk 1]
```diff
@@ -657,9 +657,19 @@ func getCredentialReport(ctx context.Context, meta schema.ClientMeta) (reportUse
...
+				// LimitExceeded is the only specific error that should not stop processing
+				// If Limit Exceeded is returned we should try and see if there is a credential report
+				// already generated so we want to sleep for 5 seconds then continue
+				if serviceError.ErrorCode() != "LimitExceeded" {
+					return nil, diag.WrapError(err)
+				}
+				time.Sleep(5 * time.Second)
 			}
-		case "ReportInProgress":
+		case "ReportInProgress", "LimitExceeded":
```
Comment (bbernays) [inline resources/services/iam/users.go:672]:
```suggestion
		case "ReportInProgress":
```

**Review by amanenk — APPROVED** [review]
LGTM

File: resources/services/iam/users.go [hunk 2]
```diff
@@ -657,7 +657,17 @@ func getCredentialReport(ctx context.Context, meta schema.ClientMeta) (reportUse
...
+				if !errors.As(err, &serviceError) {
+					return nil, diag.WrapError(err)
+				}
+				// LimitExceeded is the only specific error that should not stop processing
+				// If Limit Exceeded is returned we should try and see if there is a credential report
+				// already generated so we want to sleep for 5 seconds then continue
+				if serviceError.ErrorCode() != "LimitExceeded" {
+					return nil, diag.WrapError(err)
+				}
+				time.Sleep(5 * time.Second)
```
Comment (disq) [inline resources/services/iam/users.go:670]: Would add a `select` with `<-ctx.Done()` and a `<-time.After` to prevent 5sec hangs on interrupt.

Comment (bbernays) [reply resources/services/iam/users.go]: I updated to use the SDK helper you introduced

**Review by disq — APPROVED** [review]

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**